Application/src/visuals/ActorClass.java

```diff
@@ -0,0 +1,144 @@
+package visuals;
+
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.image.Image;
+import javafx.scene.paint.Color;
+
+/**
+ * An ActorClass wrapper modelled after DiagramClass.
+ * @author Pontus Laestadius
+ * collaborator Kosara Golemshinska
+ * @version 1.0
+ */
+public class ActorClass implements Renderable {
+
+	private Coordinates coordinates; // The coordinates of the object.
+    private int size; // Size of the Object.
+    private String name; // The class name.
+
+    // The images retrieved.
+    private static Image king1 = new Image("resources/king_cape_up.png");
+    private static Image king2 = new Image("resources/king_cape_down.png");
+
+    private static Image platform = new Image("resources/platform_default.png");
+    private static Image connector = new Image("resources/connector_default.png");
+    private static Image pillar = new Image("resources/pillar_default.png");
+
+    // Animations
+
+    // The 2 states of the king/actor.
+    private Image[] kingStates = {king1, king2};
+    // Index used to write the animation state.
+    private double kingAnimIndex = 0.0;
+
+    /**
+     * @param name the class name.
+     */
+    ActorClass(String name) {
+       this.name = name;
+   }
+
+    /**
+     * Places the ActorClass with specific coordinates and size.
+     * @param coordinates the position of the ActorClass.
+     * @param size the size for the ActorClass and it's graphics.
+     */
+    public void place(Coordinates coordinates, int size) {
+        this.coordinates = coordinates;
+        this.size = size;
+    }
+
+    /**
+     * @return the given name and not the class name of this ActorClass.
+     */
+    public String getName() {
+        String[] s = name.split(":");
+        return s[1];
+    }
+
+    /**
+     * From: Renderable
+     * Handles animation processing.
+     */
+    public void update() {
+        kingAnimIndex += 0.25;
+        if (kingAnimIndex >= kingStates.length)
+        	kingAnimIndex = 0;
+    }
+
+    /**
+     * From: Renderable
+     * @return nothing, as it's not in use.
+     */
+    public String format() {
+        return null;
+    }
+
+    /**
+     * @return the ActorClass coordinates
+     */
+    public Coordinates getCoordinates() {
+        return coordinates;
+    }
+
+    /**
+     * Renders the ActorClass on the canvas
+     * @param gc GraphicalContext for a canvas.
+     */
+    public void render(GraphicsContext gc) {
```

Comment:
shouldn't there be an @Override tag annotation here?

Reply:
Hmm it does implement Renderable but then again I'm making it similar to the DiagramClass to avoid inconsistencies, and since the latter doesn't have the @Override, I didn't include it.